react-carousel/my-app/src/components/Carousel.tsx

```diff
@@ -0,0 +1,71 @@
+import { PiCaretDoubleLeftBold, PiCaretDoubleRightBold } from 'react-icons/pi';
+import { FaCircle, FaRegCircle } from 'react-icons/fa6';
+import { Image } from '../App';
+import React, { useEffect, useState } from 'react';
+
+type Props = {
+  images: Image[];
+};
+
+export function Carousel({ images }: Props) {
+  const [selectedIndex, setSelectedIndex] = useState(0);
+
+  useEffect(() => {
+    const timeoutId = setTimeout(() => {
+      selectedIndex < images.length - 1
+        ? setSelectedIndex(selectedIndex + 1)
+        : setSelectedIndex(0);
+    }, 3000);
+    return () => clearTimeout(timeoutId);
+  }, [selectedIndex]);
+  function handleClick(event: React.MouseEvent) {
+    const target = event.currentTarget as HTMLElement;
+    switch (target.id) {
+      case 'left-caret':
+        selectedIndex > 0
+          ? setSelectedIndex(selectedIndex - 1)
+          : setSelectedIndex(images.length - 1);
+        break;
+      case 'right-caret':
+        selectedIndex < images.length - 1
+          ? setSelectedIndex(selectedIndex + 1)
+          : setSelectedIndex(0);
+        break;
+      default:
+        setSelectedIndex(parseInt(target.id));
+    }
+  }
```

Comment:
I personally would have broken this up into multiple functions. This works, but you're making this function handle logic for 3 different use cases.

I would have created a function that only handles moving forward, and another that moves backwards. Then for the dots, I would have just had them call `setSelectedIndex` directly on their event prop and pass their index